plugins/action_modify_record/src/databaseMethods.ts

```diff
@@ -1,56 +1,121 @@
+import { DBConnectType } from '../../../src/components/databaseConnect'
+
 const databaseMethods = (DBConnect: any) => ({
-  doesRecordExist: async (tableName: string, matchField: string, value: any) => {
+  getRecordId: async (tableName: string, matchField: string, value: any) => {
     const text = `
-      SELECT COUNT(*) FROM "${tableName}"
+      SELECT id FROM "${tableName}"
       WHERE "${matchField}" = $1
     `
     try {
       const result = await DBConnect.query({ text, values: [value] })
-      return result.rows[0].count > 0
+      return result?.rows[0]?.id || 0
     } catch (err) {
       console.log(err.message)
       throw err
     }
   },
-  updateRecord: async (
-    tableName: string,
-    matchField: string,
-    matchValue: any,
-    record: { [key: string]: any }
-  ) => {
+  updateRecord: async (tableName: string, id: string, record: { [key: string]: any }) => {
     const placeholders = DBConnect.getValuesPlaceholders(record)
     const matchValuePlaceholder = `$${placeholders.length + 1}`
     const text = `
       UPDATE "${tableName}" SET (${Object.keys(record)})
-      = (${DBConnect.getValuesPlaceholders(record)})
-      WHERE "${matchField}" = ${matchValuePlaceholder}
+      = (${placeholders})
+      WHERE id = ${matchValuePlaceholder}
       RETURNING *
       `
     try {
       const result = await DBConnect.query({
         text,
-        values: [...Object.values(record), matchValue],
+        values: [...Object.values(record), id],
       })
       return { success: true, [tableName]: result.rows[0] }
     } catch (err) {
       console.log(err.message)
       throw err
     }
   },
-  createRecord: async (tableName: string, record: { [key: string]: any }) => {
-    const text = `
-      INSERT INTO "${tableName}" (${Object.keys(record)}) 
-      VALUES (${DBConnect.getValuesPlaceholders(record)})
-      RETURNING *
-      `
+  createRecord: async (tableName: string, record: { [key: string]: any }) =>
+    await createRecord(tableName, record, DBConnect),
```

Comment:
Simpler to just have the local createRecord exposed here? Is there a reason why not? It would be easier to read.

Reply:
how do you mean ? Not sure how to pass through DBConnect to it (without refactoring quite a bit)

Reply:
It should be accessible since this is received by `const databaseMethods = (DBConnect: any) =>`, couldn't be in use the same way that it's done by `getRecordId` ?

Reply:
createRecord is now used in two methods, wasn't sure how to invoke it from another method (technically `this` would work but had typescript errors), so I've made it a local method that require DBConnect.  

Actually this would work: b2d0440fda951a740a1991a08a96a59683322979, retested, seems to still work

Is this what you had in mind ?